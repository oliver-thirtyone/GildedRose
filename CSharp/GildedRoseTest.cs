using System.Collections.Generic;
using ApprovalTests.Combinations;
using ApprovalTests.Reporters;
using NUnit.Framework;

namespace GildedRose
{
    [UseReporter(typeof(DiffReporter))]
    public class GildedRoseTest
    {
        [Test]
        public void foo()
        {
            var names = new List<string> {"foo", "Aged Brie", "Backstage passes to a TAFKAL80ETC concert"};
            CombinationApprovals.VerifyAllCombinations(name => DoStuff(name, 0), names);
        }

        private static string DoStuff(string name, int quality)
        {
            IList<Item> items = new List<Item> {new Item {Name = name, SellIn = 0, Quality = quality}};
            var app = new GildedRose(items);
            app.UpdateQuality();
            var result = items[0].ToString();
            return result;
        }
    }
}