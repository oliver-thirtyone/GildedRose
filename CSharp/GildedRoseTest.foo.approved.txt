[foo] => foo | -1 | 0
[Aged Brie] => Aged Brie | -1 | 2
[Backstage passes to a TAFKAL80ETC concert] => Backstage passes to a TAFKAL80ETC concert | -1 | 0
